src/view/calendar/calendar-component.jsx

```diff
@@ -235,28 +240,36 @@ export default function CalendarComponent() {
 								panel: false
 							}
 						});
-						const storeData = store.getState();
 						callbacks
 							.onSave({
-								isNew: storeData.editor.editors[editor.id]?.isNew
+								isNew: editor?.isNew
 							})
 							.then((res) => {
-								if (res?.type) {
-									const success = res.type.includes('fulfilled');
-									getBridgedFunctions().createSnackbar({
-										key: `calendar-moved-root`,
-										replace: true,
-										type: success ? 'info' : 'warning',
-										hideButton: true,
-										label: !success
-											? t('label.error_try_again', 'Something went wrong, please try again')
-											: t('message.snackbar.calendar_edits_saved', 'Edits saved correctly'),
-										autoHideTimeout: 3000
-									});
-								}
+								const success = !res.error;
+								getBridgedFunctions().createSnackbar({
+									key: `calendar-moved-root`,
+									replace: true,
+									type: success ? 'info' : 'warning',
+									hideButton: true,
+									label: !success
+										? t('label.error_try_again', 'Something went wrong, please try again')
+										: t('message.snackbar.calendar_edits_saved', 'Edits saved correctly'),
+									autoHideTimeout: 3000
+								});
 							});
+					} else {
+						getBridgedFunctions().createSnackbar({
+							key: `calendar-moved-root`,
+							replace: true,
+							type: 'warning',
+							hideButton: true,
+							label:
+								payload?.Reason?.Text ??
+								t('label.error_try_again', 'Something went wrong, please try again'),
```

Comment:
since you are handling the error ,so it will be better to show the exact error instead of a static warning "Something went wrong..."

Reply:
payload?.Reason?.Text <- this is the error, the other one is a generic fallback in case this is not usable

Reply:
Sorry my bad I didn't see it properly